nibabel/freesurfer/io.py

```diff
@@ -622,3 +624,53 @@ def _serialize_volume_info(volume_info):
             strings.append(
                 f'{key:6s} = {val[0]:.10g} {val[1]:.10g} {val[2]:.10g}\n'.encode('utf-8'))
     return b''.join(strings)
+
+def read_stats_file(file_path):
+    """Extracts stats from stats files except '*curv.stats' files
+
+    Parameters
+    ----------
+    file_path: str, required
+
+    Examples
+    --------
+    >>> stats_a2009, column_names = read_stats_file(r'lh.aparc.a2009s.stats')
+
+    """
+    with open(file_path, 'r') as f:
+        for line in f:
+            if re.findall(r'ColHeaders .*', line):
+                parameters = line.split()
+                break
+    f.close()
+    stats = np.loadtxt(file_path, comments='#', dtype=str)
+    column_names = parameters[2:]
+    return stats, column_names
```

Comment:
Absolutely. All thing works correctly except the format definition adversely affects the generality of the function. for example, 'aseg.stats' file has a different columns order compared to the 'lh.aparc.a2009s.stats'. It is while these files have the same structure.

Reply:
What about `genfromtxt`, which will detect the types for us?

```suggestion
    with open(file_path, 'r') as f:
        for line in f:
            if line.startswith('# ColHeaders '):
                columns = line.split()[2:]
                break
        return np.genfromtxt(f, dtype=None, names=columns, encoding='utf-8')
```

I just think it would be a shame to get back a numpy array of strings.